### ISSUE TITLE
Seeing issue with user-defined type

### ISSUE THREAD

**Issue by sa-mm** [MainShayne233/breakfast#12]
Following the documentation, I found an issue with user-defined types. I'm not sure how to reproduce it in a test, so I'll just explain the steps to reproduce. Suppose you have the following structure:

```elixir
# in lib/types.ex
defmodule Types do
  @type color :: :red | :green | :blue
end

# in lib/texture.ex
defmodule Texture do
  use Breakfast

  require Types

  cereal do
    field :color, Types.color()
  end
end
```

If I run:
```bash
mix clean
mix compile
```

Then, I get a compilation error in `lib/texture`. I do not see the error if I  set the type of the `:color` field to `atom()` and run

```
mix clean
mix compile
```

And then change it to `Types.color()` and compile again.

**Comment by MainShayne233** [comment]
I was able to reproduce the error! Very strange; seems like a compilation timing issue. I'll look into this; thanks for reporting!

**Comment by MainShayne233** [comment]
Ah, this one is tricky. The problem is:

- We rely on `Code.Typespec.fetch_types/1` to resolves remote types
- This function relies on the `BEAM` files being written to disk in order to get the types
- User defined types will not have their `BEAM` files written to disk until the full compilation has occured.

I think we can work around this by resolving the type from the source file, but it's a bit sketch :grimacing:

**Comment by MainShayne233** [comment]
@sa-mm I believe I might've solved the issue in `v0.1.3`. Could you give that a try?

**Comment by sa-mm** [comment]
@MainShayne233 Yup, it is working now. Thanks!
